### PR TITLE
Add default ratings to PlayerStats JSON

### PR DESCRIPTION
**Pull request by gr8st**
**SUMMARY**
This is a small code change that only saves default ratings (750) when a player starts a game. It saves two ratings:
* Rating for 'game type', such as freestyle, dropless grey, etc. Defaults to 750.
* Rating for 'overall'. This will track a player's rating across all game types. Defaults to 750.
* Note that for each rating, we also store a `last_updated` timestamp. This is in case we ever want to use that timestamp to influence rating calculation (`last_updated `values far in the past tend to mean the user's rating has less confidence).

Future code changes will begin reading ratings and using them to compute new ratings based on game result; this code change is small with the intention of verifying that we are aligned on the implementation, and specifically aligned on the new `ratings` data structure being created in the JSON schema.

**TESTING**
1. npm run clean
2. npm start   
3. http://localhost:2000 in 2 different browsers --> two different accounts
4. Play a freestyle game
5. Verify player_xx_stats.json ouput with 'freestyle' default to 750, and 'overall' default to 750
6. Play another game, this time FPS grey
7. Verify FPS grey default rating 750 is added.
8. Verify the freestyle "lastUpdated" timestamp is different than FPS Grey and Overall timestamps.

Final JSON:

[player_7de5431e-c4a0-4bf0-94e9-f8148ddcc5df_stats.json](https://github.com/pongstylin/tactics/files/14203698/player_7de5431e-c4a0-4bf0-94e9-f8148ddcc5df_stats.json)

[player_f88d0d9e-f3dd-489f-80e7-21af940cf6ce_stats.json](https://github.com/pongstylin/tactics/files/14203701/player_f88d0d9e-f3dd-489f-80e7-21af940cf6ce_stats.json)

### CODE REVIEW

File: src/models/PlayerStats.ts [hunk 1]
```diff
@@ -207,9 +229,32 @@ serializer.addType({
...
+          items: [
+            { type:'string' },
+            {
+              type:'string',
+              oneOf: [
+                {
+                  required: ['rating'],
+                  properties: {
+                    rating: { type:'number' },
+                    lastUpdated: { type:'string', subType:'Date' },
```
Comment (pongstylin) [inline src/models/PlayerStats.ts:249]: Nitpick.  The convention used elsewhere is to call the field "updatedAt".

File: src/models/PlayerStats.ts [hunk 2]
```diff
@@ -1,10 +1,13 @@
 import ActiveModel from '#models/ActiveModel.js';
 import serializer from '#utils/serializer.js';
 
+const DEFAULT_RATING = 750.0;
+
 export default class PlayerStats extends ActiveModel {
   protected data: {
     playerId: string
     stats: Map<string, any>
+    ratings: Map<string, any>
```
Comment (pongstylin) [inline src/models/PlayerStats.ts:10]: While doing it your way is acceptable, I actually intended to place the ratings in the "stats" structure.  It's admittedly confusing.  The "stats" map keys are player IDs.  If the stats are vs someone else, then the key is their player ID.  If you want your own stats, it is stored under YOUR player ID (and matches the playerId property).  I currently track the "completed" games and "abandoned" games in this location.

File: src/models/PlayerStats.ts (hunk 1 above)
Comment (gr8st) [reply src/models/PlayerStats.ts:249]: Fair enough, will update

File: src/models/PlayerStats.ts (hunk 2 above)
Comment (gr8st) [reply src/models/PlayerStats.ts:10]: Ah I see. I can do it that way, keying it off their self

**Review by gr8st — COMMENTED** [review]
Back-end rating changes

**Pull request merged**